test/tap/adduser-always-auth.js

```diff
@@ -46,19 +63,37 @@ test('npm login', function (t) {
       })
     })
 
-    var o = ''
-    var e = ''
-    var remaining = Object.keys(responses).length
-    runner.stdout.on('data', function (chunk) {
-      remaining--
-      o += chunk
-
-      var label = chunk.toString('utf8').split(':')[0]
-      runner.stdin.write(responses[label])
+    var message = 'Logged in as u on ' + common.registry + '/.'
+    runner.stdout.on('data', verifyStdout(runner, message, t))
+  })
+})
 
-      if (remaining === 0) runner.stdin.end()
+test('npm login --scope', function (t) {
```

Comment:
Also, those tests sound like great ideas. You have a few days until we're going to be in a position to merge this change, so if you want to try other approaches and refine the tests, that would be greatly appreciated. Just let us know, so we can re-review before merging. Thanks for your help!